### ISSUE TITLE
FolderExists and EnsureFolder trigger list view threshold exception on large lists

### ISSUE THREAD

**Issue by YannickRe** [pnp/pnpframework#234]
As the title says: using FolderExists and EnsureFolder in code that runs on a large list (more than 5000 items), it triggers the dreaded List View Threshold limit exceed exception.

I tracked it down to loading the FolderCollection so it can be looped to find if a folder exists yes or no:
https://github.com/pnp/pnpframework/blob/1816ab9afc9cfbb296db1dea58d63d8dafba2e7f/src/lib/PnP.Framework/Extensions/FileFolderExtensions.cs#L712

Basically, loading the FolderCollection of a large list will trigger the exception. This means that this issue isn't limited to FolderExists and EnsureFolder, but can occur everywhere that a large folder collection is loaded.
In my code, I was able to work around it using DoesFolderExists and GetFolderByServerRelativeUrl. Both don't trigger the exception, but are a bit harder to work with.

**Comment by YannickRe** [comment]
This is still a problem.

**Comment by Kizlyk** [comment]
I have the same problem, this needs to be fixed in my opinion.

**Comment by heinrich-ulbricht** [comment]
Still an exception. For me it errors here:

https://github.com/pnp/pnpframework/blob/379e91907e8a293e82ee1081100ebf5862110fb2/src/lib/PnP.Framework/Extensions/FileFolderExtensions.cs#L868-L870

This is on a list with > 5000 folders. Nothing stale here Mr. Bot ^^

This will send any PnP template provisioning files there into an error.

**Comment by heinrich-ulbricht** [comment]
Seems unlike @YannickRe I am hitting the error on getting the folder by _path_. At least for that case I have a proposed fix in my fork. But since pathes are one of my final bosses of SharePoint I am not entirely sure if this works with all kinds of URL encodings.

To reproduce it you need a folder with > 5000 subfolders which will trigger an exception at all places where the `folderCollection` is loaded.

@PaoloPia Could you re-open the issue please? It is yet to be fixed.
